nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/TestEmailNotification.java

```diff
@@ -0,0 +1,177 @@
+import org.nuxeo.mail.SmtpMailServerFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+import net.htmlparser.jericho.Renderer;
+import net.htmlparser.jericho.Source;
+
+/**
+ * Test the whole process: from subscribing to a given event until the reception of the mail notification on this event.
+ *
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ PlatformFeature.class, SmtpMailServerFeature.class })
+@Deploy("org.nuxeo.ecm.platform.notification.core")
+@Deploy("org.nuxeo.ecm.platform.notification.api")
+@Deploy("org.nuxeo.ecm.platform.url.api")
+@Deploy("org.nuxeo.ecm.platform.url.core")
+@Deploy("org.nuxeo.ecm.platform.notification.core.tests:OSGI-INF/notification-event-listener-contrib.xml")
+public class TestEmailNotification {
+
+    protected static final String DUMMY_NOTIFICATION_NAME = "DummyNotificationToSendMail";
+
+    protected static final String DUMMY_EVENT_NAME = "dummyNotificationToSendMail";
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    protected static final String ANOTHER_DOCUMENT_NAME = "anyFile2";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected NotificationManager notificationManager;
+
+    @Inject
+    protected EventService eventService;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Inject
+    protected SmtpMailServerFeature.MailsResult emailsResult;
+
+    protected DocumentModel domain;
+
+    @Before
+    public void before() {
+        domain = session.createDocumentModel("/", "domain", "Domain");
+        domain = session.createDocument(domain);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToMainDocument() {
+        // create a File document and
+        DocumentModel documentModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification
+        addSubscriptions(documentModel);
+
+        // fire the dummy event
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), documentModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(documentModel, DOCUMENT_NAME);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToParentDocument() {
+        // create a simple hierarchy (domain <- DOCUMENT_NAME <- ANOTHER_DOCUMENT_NAME)
+        DocumentModel parentDocModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification on the parent document
+        addSubscriptions(parentDocModel);
+
+        // create a child document
+        DocumentModel mainDocModel = createDocument(parentDocModel, ANOTHER_DOCUMENT_NAME);
```

Comment:
Could be placed right after `DocumentModel parentDocModel = createDocument(domain, DOCUMENT_NAME);` to clearly match the comment "create a simple hierarchy (domain <- DOCUMENT_NAME <- ANOTHER_DOCUMENT_NAME)".